src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsnonassociationsapi/service/NonAssociationMergeServiceIntTest.kt

```diff
@@ -0,0 +1,106 @@
+
+import org.assertj.core.api.Assertions
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest
+import org.springframework.context.annotation.Import
+import org.springframework.security.test.context.support.WithMockUser
+import org.springframework.test.context.transaction.TestTransaction
+import uk.gov.justice.digital.hmpps.hmppsnonassociationsapi.config.AuditorAwareImpl
+import uk.gov.justice.digital.hmpps.hmppsnonassociationsapi.config.AuthenticationFacade
+import uk.gov.justice.digital.hmpps.hmppsnonassociationsapi.helper.TestBase
+import uk.gov.justice.digital.hmpps.hmppsnonassociationsapi.jpa.repository.NonAssociationsRepository
+import java.time.LocalDateTime
+
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
+@Import(AuthenticationFacade::class, AuditorAwareImpl::class, NonAssociationsMergeService::class)
+@WithMockUser
+@DataJpaTest
+class NonAssociationMergeServiceIntTest : TestBase() {
+
+  @Autowired
+  lateinit var service: NonAssociationsMergeService
+
+  @Autowired
+  lateinit var repository: NonAssociationsRepository
+
+  @Test
+  fun testMerge() {
+    val createTime = LocalDateTime.now()
+    repository.save(
+      genNonAssociation(
+        firstPrisonerNumber = "A1234AA",
+        secondPrisonerNumber = "X1234AA",
+        createTime = createTime,
+      ),
+    )
+    repository.save(
+      genNonAssociation(
+        firstPrisonerNumber = "X1234AB",
+        secondPrisonerNumber = "A1234AA",
+        createTime = createTime,
+      ),
+    )
+    repository.save(
+      genNonAssociation(
+        firstPrisonerNumber = "B1234AA",
+        secondPrisonerNumber = "X1234AA",
+        createTime = createTime,
+      ),
+    )
+    repository.save(
+      genNonAssociation(
+        firstPrisonerNumber = "X1234AB",
+        secondPrisonerNumber = "B1234AA",
+        createTime = createTime,
+      ),
+    )
+    repository.save(
+      genNonAssociation(
+        firstPrisonerNumber = "A1234AA",
+        secondPrisonerNumber = "B1234AA",
+        createTime = createTime,
+      ),
+    )
+    repository.save(
+      genNonAssociation(
+        firstPrisonerNumber = "A1234AA",
+        secondPrisonerNumber = "C1234FF",
+        createTime = createTime,
+      ),
+    )
+    repository.save(
+      genNonAssociation(
+        firstPrisonerNumber = "D1234RR",
+        secondPrisonerNumber = "A1234AA",
+        createTime = createTime,
+      ),
+    )
+    repository.save(
+      genNonAssociation(
+        firstPrisonerNumber = "B1234AA",
+        secondPrisonerNumber = "A1234AA",
+        createTime = createTime,
+      ),
+    )
+
+    val mergedAssociations = service.mergePrisonerNumbers("A1234AA", "B1234AA")
+
+    Assertions.assertThat(mergedAssociations.toSet()).isEqualTo(
+      setOf(
```

Comment:
I imagine the `equals()` and `hashCode()` methods were added in order for this to work. I guess it's useful for testing purposes when we don't care for the specific records being updated/deleted 👍